rgdc/rgdc/rgdc.py

```diff
@@ -281,6 +286,4 @@ def search(
             params['frame_rate_min'] = frmin
             params['frame_rate_max'] = frmax
 
-        response = self.session.get('geosearch', params=params)
-        response.raise_for_status()
-        return [result for result in response.json()['results']]
+        yield from limit_offset_pager(self.session, 'geosearch', params=params)
```

Comment:
I think this is the change I'm referring to. This would turn `search()` into a generator, right? Something like
```python
result = []
for page in limit_offset_pager(self.session, 'geosearch', params=params):
    result.extend(page)
return result
```

Would preserve the original call signature and be nicer, imo.

Reply:
I think that's the only significant difference from #360

Reply:
Yes, with my change it's a generator. My reasoning was that if you wanted to use the results in a generator fashion then it's useful, and if not, you can always just cast the return of `search` to a list. If the consensus from you and @banesullivan is that it should return a list, then I'm happy to change it.

Reply:
Yeah, I'd be happy to close #360 with that change

Reply:
FWIW, another difference between this and #360 is that if a limit above the silent server limit is specified (say 1200), then in the implementation in #360, only the server limit of 1000 is returned, because pagination is only done if limit is `None`.

Reply:
I agree that returning a generator might not be very user-friendly. It may be useful to have a keyword argument `return_generator=False` that will return the generator when requested but default to casting it to a list.

I want to keep the generator under the hood in case we implement downstream functionality that could benefit from it but for most use cases right now, we'd be best off with a static list of the results.

Reply:
It looks like you both agree, so I'll change it

Reply:
Turns out it's impossible for a function to conditionally return a list / act as a generator, so I've just returned a list for now. If the need arises we can expose a new function that operates the same way but produces a generator instead. The generator functionality is all still under the hood, so nothing is really lost.

Reply:
Sounds good to me, thanks for handling this @AlmightyYakob!